tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -260,8 +260,11 @@ public function test_blocking_dependent_with_delayed_dependency( $strategy ) {
 		wp_enqueue_script( 'main-script-a3', '/main-script-a3.js', array(), null, compact( 'strategy' ) );
 		wp_enqueue_script( 'dependent-script-a3', '/dependent-script-a3.js', array( 'main-script-a3' ), null );
 		$output   = get_echo( 'wp_print_scripts' );
-		$expected = str_replace( "'", '"', "<script type='text/javascript' src='/main-script-a3.js' id='main-script-a3-js' data-wp-strategy='{$strategy}'></script>" );
-		$this->assertStringContainsString( $expected, $output, 'Blocking dependents must force delayed dependencies to become blocking.' );
+		$expected = <<<JS
+			<script type='text/javascript' src='/main-script-a3.js' id='main-script-a3-js' data-wp-strategy='{$strategy}'></script>
+			<script id="dependent-script-a3-js" src="/dependent-script-a3.js" type="text/javascript"></script>
+JS;
+		$this->assertEqualMarkup( $expected, $output, 'Blocking dependents must force delayed dependencies to become blocking.' );
```

Comment:
you are so gracious considering the HEREDOC, but now I am afraid I may have suggested something that could introduce breakage. prior to PHP 7.3 the indentation isn't removed; only the leading and trailing newline.

so here unless all of the lines in `$output` start with three tabs these might fail the comparison. for backwards compatibility it's only ugly for now 😭 (I _think_, but may be wrong, as the tests will demonstrate).

```php
		$expected = <<<JS
<script type=…>
<script id="dep…
JS;
```

Reply:
oh oops! I missed the whole `assertEqualMarkup` thing, though I don't think we can totally lean on collapsing multiple spaces, as that can introduce other breakage, right? `DOMDocument` does currently mishandle inner content in elements such as `PRE`, which truncate the initial newline if it exists.

so if we get this:

```html
<pre>
A</pre>

<pre>A</pre>
```

they are semantically equal but I think will fail `assertEqualMarkup()`

not a problem for this test case, but adding it to the HTML comparison…makes me nervous. as goofy and terrible as it sounds, I feel like I'd be more comfortable splitting each line and trimming each line then recombining before sending it to `assertEqualMarkup()` (with a comment explaining that we're removing the indentation) _or_ just stick with the ugly code for now where it's not a problem.

Reply:
Since `assertEqualMarkup` here is limited to the tests for scripts, I'm not concerned about `pre`. If anything the phpdoc description could be updated to note that it normalizes whitespace, including for elements that preserve whitespace.

Reply:
yeah I guess if this breaks tests then someone will know immediately, though only after being puzzled by why their valid code fails. note that my comment is not about preserving whitespace but the contrary. in the case of `PRE` it is one of the few elements that strips away whitespace.

oh interesting - I missed that this is all limited to the scripts tests file and isn't modifying the base `assertEqualMarkup` - all good!